src/crimson/net/Protocol.cc

```diff
@@ -69,12 +70,12 @@ void Protocol::close(bool dispatch_reset,
     socket->shutdown();
   }
   set_write_state(write_state_t::drop);
-  auto gate_closed = pending_dispatch.close();
+  auto gate_closed = close_dispatch_gate();
   auto reset_dispatched = seastar::futurize_invoke([this, dispatch_reset, is_replace] {
     if (dispatch_reset) {
-      return dispatcher.ms_handle_reset(
-          seastar::static_pointer_cast<SocketConnection>(conn.shared_from_this()),
-          is_replace);
+      dispatcher.ms_handle_reset(
+	  seastar::static_pointer_cast<SocketConnection>(conn.shared_from_this()),
+	  is_replace);
```

Comment:
I'm worried about this.  Letting this handler set run asyncronously risks the event from the connection reopening being processed first.  That seems bad.

Reply:
i have the same concern. @cyx1231st what do you think?

Reply:
@tchaikov see https://github.com/ceph/ceph/pull/34492#discussion_r428401563

I think it's good, our current decision is to make "reset" a pure synchronous event which returns void, and `ChainedDispatchers` will also dispatch it synchronously. The order is enforced as the connect/accept/remote-reset/reset events are dispatched at real time.